src/main-oe/sgxlkl_params.c

```diff
@@ -38,285 +43,107 @@ struct sgxlkl_config_elem
     };
 };
 
+typedef struct json_callback_data
+{
+} json_callback_data_t;
+
+#define DEFAULT_SGXLKL_VERBOSE 0
+#define DEFAULT_SGXLKL_CWD "/"
+#define DEFAULT_SGXLKL_GW4 "10.0.1.254"
+/* The default heap size will only be used if no heap size is specified and
+ * either we are in simulation mode, or we are in HW mode and a key is provided
+ * via SGXLKL_KEY.
+ */
+#define DEFAULT_SGXLKL_OE_HEAP_PAGE_COUNT 8192 /* 8192 * 4K = 32MB */
+#define DEFAULT_SGXLKL_HEAP_SIZE 200 * 1024 * 1024
+#define DEFAULT_SGXLKL_HOSTNAME "lkl"
+#define DEFAULT_SGXLKL_IP4 "10.0.1.1"
+#define DEFAULT_SGXLKL_MASK4 "24"
+#define DEFAULT_SGXLKL_MAX_USER_THREADS 256
+#define DEFAULT_SGXLKL_ESLEEP 16000
+#define DEFAULT_SGXLKL_ETHREADS 1
+#define DEFAULT_SGXLKL_ESPINS 500
+#define DEFAULT_SGXLKL_STACK_SIZE 512 * 1024
+#define DEFAULT_SGXLKL_SWIOTLB 1
+#define DEFAULT_SGXLKL_TAP "sgxlkl_tap0"
+#define DEFAULT_SGXLKL_WG_IP "10.0.2.1"
+#define DEFAULT_SGXLKL_WG_PORT 56002
+#define DEFAULT_SGXLKL_KERNEL_CMD "mem=32M"
+#define DEFAULT_SGXLKL_HOSTNET false
+#define DEFAULT_SGXLKL_TAP_MTU 0
```

Comment:
Shouldn't this live in the header file with the other parameters?

Reply:
No, these are now completely local to the params, so they are in the right file. The default enclave config is specified in the auto-generated files, but we can't use them in `static struct sgxlkl_config_elem sgxlkl_config[] = ...` because they are not proper constant literals. Note that these defaults are not actually used at runtime; this whole config construct is used only for debug-mode overrides via environment variables, but only if their settings have been configured, i.e. the default is never used. I'd really rather get rid of the environment variables altogether...